src/libdredd/src/mutate_visitor.cc

```diff
@@ -278,6 +280,38 @@ bool MutateVisitor::VisitBinaryOperator(
   return true;
 }
 
+bool MutateVisitor::VisitExpr(clang::Expr* expr) {
+  if (!IsInFunction()) {
+    // Only consider mutating expressions that occur inside functions.
+    return true;
+  }
+
+  // Unary and binary operators are intercepted separately.
+  if (llvm::dyn_cast<clang::BinaryOperator>(expr) != nullptr ||
+      llvm::dyn_cast<clang::UnaryOperator>(expr) != nullptr) {
+    return true;
+  }
+
+  // There is no useful way to mutate L-Value boolean expressions.
+  if (expr->isLValue() && expr->getType()->isBooleanType()) {
+    return true;
+  }
+
+  // We only want to mutate L-Values in C++ mode.
+  if (expr->isLValue() &&
+      !compiler_instance_.getASTContext().getLangOpts().CPlusPlus) {
+    return true;
+  }
```

Comment:
Is this true? I can't remember quite when things are l-values in C - but do you have an example of something that's an l-value in C that we specifically don't want to mutate? Is the point that no basic operators in C return l-values? If you could elaborate on the comment a little that would be good. (Or if it turns out there are legit cases for l-values being mutated in C, please revisit.)

Reply:
I was incorrect so have changed it to mutate l-values in C.